rules/prefer-starts-ends-with.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const doesNotContain = (string, chars) => chars.every(char => !string.includes(char));
+
+const isSimpleString = string => doesNotContain(
+	string,
+	['^', '$', '+', '[', '{', '(', '\\', '.', '?', '*']
```

Comment:
Thanks, @firasdib but I am not sure if that's exactly what we can use here. Please allow me to clarify. We want to catch the case that can be replaced with `string#startsWith` and `string#endsWith`, and the way I am approaching it is to take the test string, check if the first character is '$' and then check if the rest of the string is a _simple_ (and use similar logic for checking `endsWith`). It boils down to how to tell if that rest part of the test string is a _simple_ string. By _simple_ string I mean it doesn't use regex special characters (because otherwise it is a legit use of regex and can't be replaced with `string#startsWith`). Here is some examples of what I'd consider to be _simple_ strings:
```
foo
\^foo
!!!
\$
我
```

and some examples of NOT _simple_ strings
```
fo.o
foo+
[f,g]oo
\wfoo
a\b
$
```
I am not sure how we can modify the regex to tell these groups apart. I updated the test cases on https://regex101.com/r/DPdFPg/4/tests